src/main/java/com/atlassian/bitbucket/jenkins/internal/model/deployment/BitbucketDeploymentEnvironment.java

```diff
@@ -0,0 +1,144 @@
+package com.atlassian.bitbucket.jenkins.internal.model.deployment;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.StringUtils.stripToNull;
+
+/**
+ * The details of an environment that was deployed to.
+ *
+ * @since deployments
+ */
+public final class BitbucketDeploymentEnvironment {
+
+    private static final String DISPLAY_NAME = "displayName";
+    private static final String KEY = "key";
+    private static final String TYPE = "type";
+    private static final String URL = "url";
+
+    private final String key;
+    private final String name;
+    private final String type;
+    private final String url;
+
+    private BitbucketDeploymentEnvironment(Builder builder) {
+        key = builder.key;
+        name = builder.name;
+        type = builder.type == null ? null : builder.type.name();
+        url = builder.url;
+    }
+
+    @JsonCreator
+    public BitbucketDeploymentEnvironment(@JsonProperty(KEY) String key,
+                                          @JsonProperty(DISPLAY_NAME) String name,
+                                          @JsonProperty(TYPE) String type,
+                                          @JsonProperty(URL) String url) {
+        this.key = key;
+        this.name = name;
+        this.type = type;
+        this.url = url;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I see you've added `equals` and `hashCode` for all entity classes, is there a reason? I'm slightly concerned that we need to add a field and forget to update `equals/hashCode` when we do. I vote for removing them unless you need them for some reason?